src/services/TimerService.ts

```diff
@@ -0,0 +1,149 @@
+import { Service } from "../utils/Service";
+
+/**
+ * A function to execute when a {@link TimerEvent} fires.
+ */
+export type TimerHandler = (d?: Date) => void;
+
+/**
+ * A map to hold a list of {@link TimerHandler} functions to run when each {@link TimerEvent} fires.
+ */
+export type TimerHandlerMap = Map<string, TimerHandler[]>;
+
+/**
+ * Various timer-related events that can be listened for.
+ */
+export enum TimerEvent {
+    SECOND = "second",
+    START = "start",
+    STOP = "stop"
+}
+
+/**
+ * String representation of TimerEvents.
+ */
+export type TimerEventStr = "start" | "second" | "stop";
+
+/**
+ * Timer service: fires {@link TimerEvent}s when appropriate.
+ *
+ * Register a {@link TimerHandler} to listen for {@link TimerEvent}s by using {@link on}
+ */
+export class TimerService extends Service {
+
+    private static readonly INSTANCE: TimerService = new TimerService();
```

Comment:
I understand the need for some type of global state for something that provides a tick, but Services aren't meant to be singletons. Instead, they're meant to be dynamically-created instances that are dependency-injected into a UIScreen for things like database access. For example:
`this.getService(PasscodeService)`
This way you don't have to expose direct database access to the screen's code and there's proper separation of concerns between the view and domain logic.

So while this _logically_ works as a Service, because it has to have some global state, I don't think it'd be appropriate to have this class as a Service. Maybe it'd be better as a utility instead? Thoughts?

Reply:
I have no idea how I missed that design in `UIScreen`. Yeah, I agree there's a better way to do it, let me play around for a bit. Utility makes sense, and is the only thing I can think of at the moment that would make sense. I'm not sure of the implications of making it an instance field belonging to the app, with respect to accessing the global timer instance.

Reply:
Thanks for the clarification on the design btw!

Reply:
Oh, that's actually a good point. Do we know how Expo handles timeouts when the screen is locked or the app is closed?

Reply:
We don't have to have the app run in the background, but is the timer correct when the app is resumed?

Reply:
I'll take a look and see. Expo provides a mixin for classes that use timeouts in order to avoid updating UI elements that don't exist (e.g. have been unmounted), but for ES6 classes we'd need to use a third-party mixin per their documentation. I'll find a link shortly.

Reply:
https://docs.expo.io/versions/latest/react-native/timers/#timermixin

Reply:
Nevermind, just answered that question myself in click-through testing :p It does work correctly!